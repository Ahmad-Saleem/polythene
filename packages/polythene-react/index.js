export * from "polythene-react-base";
export * from "polythene-react-button";
export * from "polythene-react-card";
export * from "polythene-react-checkbox";
export * from "polythene-react-dialog";
export * from "polythene-react-dialog-pane";
export * from "polythene-react-fab";
export * from "polythene-react-icon";
export * from "polythene-react-icon-button";
export * from "polythene-react-list";
export * from "polythene-react-list-tile";
export * from "polythene-react-menu";
export * from "polythene-react-radio-button";
export * from "polythene-react-raised-button";
export * from "polythene-react-ripple";
export * from "polythene-react-shadow";
export * from "polythene-react-svg";
export * from "polythene-react-switch";
export * from "polythene-react-textfield";
export * from "polythene-react-toolbar";
