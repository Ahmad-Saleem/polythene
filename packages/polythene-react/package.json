{
  "name": "polythene-react",
  "version": "1.0.0-rc.8",
  "description": "",
  "main": "dist/polythene-react.js",
  "module": "dist/polythene-react.mjs",
  "scripts": {
    "clean": "rm -rf dist && mkdir dist",
    "rollup": "../../node_modules/rollup/bin/rollup -c ../../scripts/rollup.umd.js && ../../node_modules/rollup/bin/rollup -c ../../scripts/rollup.es.js",
    "build-standalone": "DEPS=1 ENTRY=scripts/standalone.js SOURCEMAP=1 DEST=./dist/polythene-react-standalone.js ../../node_modules/rollup/bin/rollup -c ../../scripts/rollup.umd-includes.js",
    "build": "yarn run rollup && yarn run build-standalone"
  },
  "files": [
    "dist",
    "README.md"
  ],
  "dependencies": {
    "polythene-css": "^1.0.0-rc.8",
    "polythene-react-base": "^1.0.0-rc.8",
    "polythene-react-button": "^1.0.0-rc.8",
    "polythene-react-card": "^1.0.0-rc.8",
    "polythene-react-checkbox": "^1.0.0-rc.8",
    "polythene-react-dialog": "^1.0.0-rc.8",
    "polythene-react-dialog-pane": "^1.0.0-rc.8",
    "polythene-react-fab": "^1.0.0-rc.8",
    "polythene-react-icon": "^1.0.0-rc.8",
    "polythene-react-icon-button": "^1.0.0-rc.8",
    "polythene-react-ios-spinner": "^1.0.0-rc.8",
    "polythene-react-list": "^1.0.0-rc.8",
    "polythene-react-list-tile": "^1.0.0-rc.8",
    "polythene-react-material-design-progress-spinner": "^1.0.0-rc.8",
    "polythene-react-material-design-spinner": "^1.0.0-rc.8",
    "polythene-react-menu": "^1.0.0-rc.8",
    "polythene-react-notification": "^1.0.0-rc.8",
    "polythene-react-radio-button": "^1.0.0-rc.8",
    "polythene-react-radio-group": "^1.0.0-rc.8",
    "polythene-react-raised-button": "^1.0.0-rc.8",
    "polythene-react-ripple": "^1.0.0-rc.8",
    "polythene-react-search": "^1.0.0-rc.8",
    "polythene-react-shadow": "^1.0.0-rc.8",
    "polythene-react-slider": "^1.0.0-rc.8",
    "polythene-react-snackbar": "^1.0.0-rc.8",
    "polythene-react-svg": "^1.0.0-rc.8",
    "polythene-react-switch": "^1.0.0-rc.8",
    "polythene-react-tabs": "^1.0.0-rc.8",
    "polythene-react-textfield": "^1.0.0-rc.8",
    "polythene-react-toolbar": "^1.0.0-rc.8"
  },
  "devDependencies": {
    "polythene-core": "^1.0.0-rc.8",
    "polythene-core-css": "^1.0.0-rc.8",
    "polythene-fastclick": "^1.0.0-rc.8",
    "react": "^15.6.1",
    "react-hyperscript": "^3.0.0"
  },
  "author": "Arthur Clemens <arthurclemens@gmail.com> (http://arthurclemens.com)",
  "license": "MIT"
}
