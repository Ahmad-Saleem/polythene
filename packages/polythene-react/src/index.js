export * from "./core";
export { button } from "./button";
export { icon } from "./icon";
export { ripple } from "./ripple";
export { shadow } from "./shadow";
export { svg } from "./svg";
