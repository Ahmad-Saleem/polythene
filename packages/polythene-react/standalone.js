export * from "./index";
import { addLayoutStyles } from "polythene-utilities";
import { addTypography, addRoboto } from "polythene-style";
import { addFastClick } from "polythene-fastclick";

addTypography();
addRoboto();
addLayoutStyles();
addFastClick();
