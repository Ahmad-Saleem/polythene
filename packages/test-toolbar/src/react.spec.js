import { runSnapshots } from "../../../scripts/tests/react-snapshots";
import { renderer, Toolbar, ToolbarTitle } from "polythene-react";
import specTests from "./spec-tests.js";
import reactTests from "./tests-react.js";

runSnapshots({
  tests: specTests({ Toolbar, ToolbarTitle, renderer }).concat(reactTests),
  renderer
});
