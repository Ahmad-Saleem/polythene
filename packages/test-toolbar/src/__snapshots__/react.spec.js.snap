exports[`test Child node 1`] = `
<div
  className="pe-toolbar    ">
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\"/></svg>",
                }
          } />,
      }
    } />
  <div
    className="pe-toolbar__title">
    Toolbar with a very very very very very very very very very long title
  </div>
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z\"/></svg>",
                }
          } />,
      }
    } />
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\"/></svg>",
                }
          } />,
      }
    } />
</div>
`;

exports[`test Child node, option compact (JSX) 1`] = `
<Toolbar
  compact={true}>
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\"/></svg>",
                }
          } />,
      }
    } />
  <div
    className="pe-toolbar__title">
    Toolbar with a very very very very very very very very very long title
  </div>
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z\"/></svg>",
                }
          } />,
      }
    } />
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\"/></svg>",
                }
          } />,
      }
    } />
</Toolbar>
`;

exports[`test Dark tone class + light tone 1`] = `
<div
  className="pe-toolbar   pe-light-tone ">
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\"/></svg>",
                }
          } />,
      }
    } />
  <div
    className="pe-toolbar__title">
    Toolbar with a very very very very very very very very very long title
  </div>
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z\"/></svg>",
                }
          } />,
      }
    } />
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\"/></svg>",
                }
          } />,
      }
    } />
</div>
`;

exports[`test Dark tone class + light tone class 1`] = `
<div
  className="pe-toolbar    pe-light-tone">
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\"/></svg>",
                }
          } />,
      }
    } />
  <div
    className="pe-toolbar__title">
    Toolbar with a very very very very very very very very very long title
  </div>
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z\"/></svg>",
                }
          } />,
      }
    } />
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\"/></svg>",
                }
          } />,
      }
    } />
</div>
`;

exports[`test JSX toolbar row (JSX) 1`] = `
<Toolbar>
  <ToolbarButton
    svg={
      <svg
        height="24"
        viewBox="0 0 24 24"
        width="24">
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z" />
      </svg>
    } />
  <span>
    Title
  </span>
  <ToolbarButton
    svg={
      <svg
        height="24"
        viewBox="0 0 24 24"
        width="24">
        <path
          d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z" />
      </svg>
    } />
  <ToolbarButton
    svg={
      <svg
        height="24"
        viewBox="0 0 24 24"
        width="24">
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z" />
      </svg>
    } />
</Toolbar>
`;

exports[`test Option: compact 1`] = `
<div
  className="pe-toolbar pe-toolbar--compact   ">
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\"/></svg>",
                }
          } />,
      }
    } />
  <div
    className="pe-toolbar__title">
    Toolbar with a very very very very very very very very very long title
  </div>
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z\"/></svg>",
                }
          } />,
      }
    } />
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\"/></svg>",
                }
          } />,
      }
    } />
</div>
`;

exports[`test Option: content -- dark tone class 1`] = `
<div
  className="pe-toolbar    ">
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\"/></svg>",
                }
          } />,
      }
    } />
  <div
    className="pe-toolbar__title">
    Toolbar with a very very very very very very very very very long title
  </div>
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z\"/></svg>",
                }
          } />,
      }
    } />
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\"/></svg>",
                }
          } />,
      }
    } />
</div>
`;

exports[`test Option: content 1`] = `
<div
  className="pe-toolbar    ">
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\"/></svg>",
                }
          } />,
      }
    } />
  <div
    className="pe-toolbar__title">
    Toolbar with a very very very very very very very very very long title
  </div>
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z\"/></svg>",
                }
          } />,
      }
    } />
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\"/></svg>",
                }
          } />,
      }
    } />
</div>
`;

exports[`test Option: shadow (JSX) 1`] = `
<div
  style={
    Object {
      "position": "relative",
    }
  }>
  <Toolbar>
    <ToolbarButton
      svg={
        <svg
          height="24"
          viewBox="0 0 24 24"
          width="24">
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z" />
        </svg>
      } />
    <span>
      Title
    </span>
    <ToolbarButton
      svg={
        <svg
          height="24"
          viewBox="0 0 24 24"
          width="24">
          <path
            d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z" />
        </svg>
      } />
    <ToolbarButton
      svg={
        <svg
          height="24"
          viewBox="0 0 24 24"
          width="24">
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z" />
        </svg>
      } />
  </Toolbar>
  <Shadow />
</div>
`;

exports[`test Option: shadow 1`] = `
<div
  style={
    Object {
      "position": "relative",
    }
  }>
  <Toolbar>
    <IconButton
      icon={
        Object {
          "svg": <div
            dangerouslySetInnerHTML={
                Object {
                    "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\"/></svg>",
                  }
            } />,
        }
      } />
    <div
      className="pe-toolbar__title">
      Toolbar with a very very very very very very very very very long title
    </div>
    <IconButton
      icon={
        Object {
          "svg": <div
            dangerouslySetInnerHTML={
                Object {
                    "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z\"/></svg>",
                  }
            } />,
        }
      } />
    <IconButton
      icon={
        Object {
          "svg": <div
            dangerouslySetInnerHTML={
                Object {
                    "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\"/></svg>",
                  }
            } />,
        }
      } />
  </Toolbar>
  <Shadow />
</div>
`;

exports[`test Option: style (colors and height) (JSX) 1`] = `
<Toolbar
  content={
    Array [
      <IconButton
        icon={
            Object {
                "svg": <div
                  dangerouslySetInnerHTML={
                        Object {
                              "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\"/></svg>",
                            }
                  } />,
              }
        } />,
      <div
        className="pe-toolbar__title">
        Toolbar with a very very very very very very very very very long title
    </div>,
      <IconButton
        icon={
            Object {
                "svg": <div
                  dangerouslySetInnerHTML={
                        Object {
                              "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z\"/></svg>",
                            }
                  } />,
              }
        } />,
      <IconButton
        icon={
            Object {
                "svg": <div
                  dangerouslySetInnerHTML={
                        Object {
                              "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\"/></svg>",
                            }
                  } />,
              }
        } />,
    ]
  }
  style={
    Object {
      "backgroundColor": "#EF6C00",
      "color": "#fff",
      "height": "72px",
    }
  } />
`;

exports[`test Option: style (colors and height) 1`] = `
<div
  className="pe-toolbar    "
  style={
    Object {
      "backgroundColor": "#EF6C00",
      "color": "#fff",
      "height": "72px",
    }
  }>
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\"/></svg>",
                }
          } />,
      }
    } />
  <div
    className="pe-toolbar__title">
    Toolbar with a very very very very very very very very very long title
  </div>
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z\"/></svg>",
                }
          } />,
      }
    } />
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\"/></svg>",
                }
          } />,
      }
    } />
</div>
`;

exports[`test Shadow 1`] = `
<div
  style={
    Object {
      "position": "relative",
    }
  }>
  <Toolbar>
    <IconButton
      icon={
        Object {
          "svg": <div
            dangerouslySetInnerHTML={
                Object {
                    "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\"/></svg>",
                  }
            } />,
        }
      } />
    <div
      className="pe-toolbar__title">
      Toolbar with a very very very very very very very very very long title
    </div>
    <IconButton
      icon={
        Object {
          "svg": <div
            dangerouslySetInnerHTML={
                Object {
                    "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z\"/></svg>",
                  }
            } />,
        }
      } />
    <IconButton
      icon={
        Object {
          "svg": <div
            dangerouslySetInnerHTML={
                Object {
                    "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\"/></svg>",
                  }
            } />,
        }
      } />
  </Toolbar>
  <Shadow />
</div>
`;

exports[`test Themed (dark tone) 1`] = `
<div
  className="pe-toolbar  pe-dark-tone  tests-toolbar-themed-toolbar">
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\"/></svg>",
                }
          } />,
      }
    } />
  <div
    className="pe-toolbar__title">
    Toolbar with a very very very very very very very very very long title
  </div>
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z\"/></svg>",
                }
          } />,
      }
    } />
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\"/></svg>",
                }
          } />,
      }
    } />
</div>
`;

exports[`test Themed -- dark tone class 1`] = `
<div
  className="pe-toolbar    tests-toolbar-themed-toolbar">
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\"/></svg>",
                }
          } />,
      }
    } />
  <div
    className="pe-toolbar__title">
    Toolbar with a very very very very very very very very very long title
  </div>
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z\"/></svg>",
                }
          } />,
      }
    } />
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\"/></svg>",
                }
          } />,
      }
    } />
</div>
`;

exports[`test Title as span 1`] = `
<div
  className="pe-toolbar    ">
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\"/></svg>",
                }
          } />,
      }
    } />
  <span>
    Toolbar with a very very very long title
  </span>
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\"/></svg>",
                }
          } />,
      }
    } />
</div>
`;

exports[`test Title as span, indented (without left icon) 1`] = `
<div
  className="pe-toolbar    ">
  <div
    className="pe-toolbar__title--indent">
    Indented title
  </div>
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\"/></svg>",
                }
          } />,
      }
    } />
</div>
`;

exports[`test ToolbarTitle 1`] = `
<div
  className="pe-toolbar    ">
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\"/></svg>",
                }
          } />,
      }
    } />
  <ToolbarTitle
    text="Toolbar with a very very very long title" />
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\"/></svg>",
                }
          } />,
      }
    } />
</div>
`;

exports[`test ToolbarTitle, at start 1`] = `
<div
  className="pe-toolbar    ">
  <ToolbarTitle
    text="Toolbar with a very very very long title" />
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\"/></svg>",
                }
          } />,
      }
    } />
</div>
`;

exports[`test ToolbarTitle, centered 1`] = `
<div
  className="pe-toolbar    ">
  <ToolbarTitle
    center={true}
    text="Toolbar with a very very very long title" />
</div>
`;

exports[`test ToolbarTitle, centered, balanced with a placeholder 1`] = `
<div
  className="pe-toolbar    ">
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\"/></svg>",
                }
          } />,
      }
    } />
  <ToolbarTitle
    center={true}
    text="Toolbar with a very very very long title" />
  <IconButton
    icon={
      Object {
        "content": "",
      }
    }
    inactive={true} />
</div>
`;

exports[`test ToolbarTitle, indented (without left icon) 1`] = `
<div
  className="pe-toolbar    ">
  <ToolbarTitle
    indent={true}
    text="Toolbar with a very very very long title" />
  <IconButton
    icon={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\"/></svg>",
                }
          } />,
      }
    } />
</div>
`;
